### DISCUSSION TITLE
[V3] Using a layout for placeholder() on a full-page component

### DISCUSSION THREAD

**Discussion by gazzoy** [livewire/livewire#7107] [1. Help] [answered]
I'm trying to specify a layout for placeholder() on a full-page component, but no luck. Is this supported?

As follows, I'm trying to render placeholder() with `layouts.admin`, but it always renders with `layouts.app`. What am I missing? Plz ask me any questions.

web.php
```
Route::get('/do-something', DoSomething::class)->lazy()->name('do_something');
```

config/livewire.php
```
'layout' => 'layouts.app',
```

DoSomething.php
```
class DoSomething extends Component
{
...
    public function placeholder(array $params = [])
    {
        return view('livewire.placeholders.skeleton', $params)->layout("layouts.admin"); // this does NOT work
    }
...
}
```

**Comment by gazzoy** [comment]
Could anyone help please?

**Comment by lrljoe** [comment]
The placeholder just replaces your component with the placeholder.

You cannot swap layouts at that stage.  It replaces the component, within the layout, until the lifecycle completes.

Your placeholder must also have the same parent element type as your component.

What are you trying to achieve here?

If you want a wholly different display while loading, look at full screen modals instead.

**Reply by adriansio** [reply]
Just like me, he wants to display a component that has a custom layout, and he wants to load it lazily.
Any component should be loaded lazily, it just happens that some have a custom layout instead of the default one.

**Reply by lrljoe** [reply]
This is not what Layouts are for... I suggest you review the Laravel guidance on Layouts.

**Reply by gazzoy** [reply]
@adriansio Thanks for you comments. I too think any component should be loaded lazily.

@lrljoe Let me rephrase that. I tried below code, which is simply specifies a custom layout for a full-page-component, but it also does not work either. Is this out of the way?

```
#[Layout('layouts.admin')] // this does NOT work either
class DoSomething extends Component
{
...
    public function placeholder(array $params = [])
    {
        return view('livewire.placeholders.skeleton', $params) 
    }
...
}
```

**Reply by lrljoe** [reply, marked as answer]
Just did some testing on my end, and have had a chat with the core team.  This is a known bug.  I've had a quick look at the code, and it won't be a super simple fix sadly!

Please can I very politely suggest that you add your detail to #6907, and close this discussion, so that we don't end up with 8 discussions on one issue.

**Reply by gazzoy** [reply]
@lrljoe Many thanks for supporting this. It's very good to know this is a known bug. Sure, will close this discussion, then add my detail on #6907.
